docs/faq.rst

```diff
@@ -186,6 +189,46 @@ Yes, if you want to detect accidental data damage (like bit rot), use the
 If you want to be able to detect malicious tampering also, use an encrypted
 repo. It will then be able to check using CRCs and HMACs.
 
+.. _faq-integrityerror:
+
+I get an IntegrityError or similar - what now?
+----------------------------------------------
+
+The first step should be to check whether it's a problem with the disk drive,
+IntegrityErrors can be a sign of drive failure or other hardware issues.
+
+Using the smartmontools one can retrieve self-diagnostics of the drive in question
+(where the repository is located, use *findmnt*, *mount* or *lsblk* to find the
+*/dev/...* path of the drive)::
+
+    # smartctl -a /dev/sdSomething
+
+Attributes that are a typical cause of data corruption are *Offline_Uncorrectable*,
+*Current_Pending_Sector*, *Reported_Uncorrect*. A high *UDMA_CRC_Error_Count* usually
+indicates a bad cable. If the *entire drive* is failing, then all data should be copied
+off it as soon as possible.
+
+Some drives log IO errors, which are also logged by the system (refer to the journal/dmesg).
+IO errors that impact only the filesystem can go unnoticed, since they are not reported
+to applications (e.g. Borg), but can still corrupt data.
+
+If any of these are suspicious, a self-test is recommended::
+
+    # smartctl -t long /dev/sdSomething
```

Comment:
Hmm, if I see the drive is bad from -a, I do not need to run the long test, I'ld rather quickly make sure I have all data backed up and get a new drive.

If the SMART data tells "all ok", then running a long test to find if that is really true for the complete media surface is interesting, errors on unused sectors could be discovered then.